### PR TITLE
chore: migrate nose to pytest

### PR DESCRIPTION
**Pull request by zhangcheng**
Since nose is too outdated to support Python 3.10
```
  File "/Users/czhang/workspace/vendor/flask_injector/ve3.10/lib/python3.10/site-packages/nose/suite.py", line 106, in _set_tests
    if isinstance(tests, collections.Callable) and not is_suite:
AttributeError: module 'collections' has no attribute 'Callable'
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,8 +5,7 @@ ci: test lint
 
 .PHONY: test
 test:
-	nosetests -v $(SOURCES) \
-		--with-coverage --cover-package=flask_injector --cover-html --cover-branches --cover-xml
+	coverage run --source=$(SOURCES) -m pytest -v $(SOURCES)/tests.py && coverage report -m
```
Comment (jstasiak) [inline Makefile:8]: The `$(SOURCES/tests.py` part depends on `$SOURCES` being set to a very specific value – namely just `flask_injector`. I'm not happy with depending on that so I propose to use just `pytest -v $(SOURCES)`

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 black;implementation_name=="cpython"
+coverage
```
Comment (jstasiak) [inline requirements-dev.txt:2]:
```suggestion
```

#70 handles this

File: .gitignore [hunk 3]
```diff
@@ -4,4 +4,6 @@ build/
 *.egg-info/
 .mypy_cache/
 cover/
+.coverage
 coverage.xml
+ve*/
```
Comment (jstasiak) [inline .gitignore:9]: Please remove the additions to `.gitignore`, they aren't related to this PR

**Review by jstasiak — CHANGES_REQUESTED** [review]
Hey! Thank you very much for this, there are a few small changes to make but LGTM otherwise

**Comment by zhangcheng** [comment]
> Hey! Thank you very much for this, there are a few small changes to make but LGTM otherwise

Got it @jstasiak , will take care of them in a while. :-)

**Comment by jstasiak** [comment]
Thank you for the contribution, merged as #73 with some changes

**Pull request closed without merging**